src/main/scala/prci/ClockParameters.scala

```diff
@@ -83,25 +86,32 @@ case class ClockGroupDriverParameters(
   num: Int = 1,
   driveFn: ClockGroupDriver.DriveFn = ClockGroupDriver.driveFromImplicitClock
 ) {
-  def drive(node: ClockGroupEphemeralNode)(implicit p: Parameters, vn: ValName): ModuleValue[HeterogeneousBag[ClockGroupBundle]] = {
+  def drive(node: ClockGroupEphemeralNode)(implicit p: Parameters, vn: ValName): ModuleValue[RecordListMap[ClockGroupBundle]] = {
     driveFn(node, num, p, vn)
   }
 }
 
 object ClockGroupDriver {
-  type DriveFn = (ClockGroupEphemeralNode, Int, Parameters, ValName) => ModuleValue[HeterogeneousBag[ClockGroupBundle]]
+  type DriveFn = (ClockGroupEphemeralNode, Int, Parameters, ValName) => ModuleValue[RecordListMap[ClockGroupBundle]]
 
   def driveFromImplicitClock: DriveFn = { (groups, num, p, vn) =>
     implicit val pp = p
     val dummyClockGroupSourceNode: ClockGroupSourceNode = SimpleClockGroupSource(num)
     groups :*= dummyClockGroupSourceNode
-    InModuleBody { HeterogeneousBag[ClockGroupBundle](Nil) }
+    InModuleBody { RecordListMap[ClockGroupBundle](ListMap()) }
   }
 
-  def driveFromIOs: DriveFn = { (groups, num, p, vn) =>
+  def driveFromIOs()(implicit valName: ValName): DriveFn = { (groups, num, p, vn) =>
     implicit val pp = p
     val ioClockGroupSourceNode = ClockGroupSourceNode(List.fill(num) { ClockGroupSourceParameters() })
     groups :*= ioClockGroupSourceNode
-    InModuleBody { ioClockGroupSourceNode.makeIOs()(vn) }
+    InModuleBody {
+      val bundlesAndEdges = ioClockGroupSourceNode.out
```

Comment:
this is basically a rewrite of `makeIOs` from diplomacy.Nodes, which assumes Heterogeneous Bag

Reply:
Yeah in a follow up PR we should just entirely replace that use of `HeterogeneousBag` with `RecordListMap`, I think. Will require sed/updating a moderate amount of boilerplate closed source code to change arguments like "`io: HeterogeneousBag`" to "`io: Record`" though